src/controller/tests/TestWriteChunking.cpp

```diff
@@ -93,10 +99,57 @@ class TestWriteChunking : public Test::AppContext
         std::vector<bool> expectedStatus;
     };
 
-    void RunTest(Instructions instructions);
-    void RunTest_NonEmptyReplaceAll(Instructions instructions);
+    void RunTest(Instructions instructions, EncodingMethod encodingMethod);
+    void RunTest_NonEmptyReplaceAll(Instructions instructions, EncodingMethod encodingMethod);
+
+    template <class T>
+    void EncodeAttributeListIntoTLV(const DataModel::List<T> & aListAttribute, TLV::ScopedBufferTLVReader & outTlvReader);
 };
 
+// Encodes an attribute of List Data Type into a TLV Reader object for testing WriteClient::PutPreencodedAttribute
+// Warning: This method only encodes uint8_t or 1-octet ByteSpans
+template <class T>
+void TestWriteChunking::EncodeAttributeListIntoTLV(const DataModel::List<T> & aListAttribute,
+                                                   TLV::ScopedBufferTLVReader & outEncodedListTlvReader)
+{
+    static_assert(std::is_same<T, chip::ByteSpan>::value || std::is_same<T, uint8_t>::value,
+                  "This method only encodes uint8_t or 1-octet ByteSpans");
+
+    size_t estimatedSize = 0;
+    for (size_t i = 0; i < aListAttribute.size(); i++)
+    {
+        if constexpr (std::is_same<T, uint8_t>::value)
+        {
+            // Control Octet (1) + size of uint8_t
+            estimatedSize += 1 + sizeof(uint8_t);
+        }
+        else if constexpr (std::is_same<T, chip::ByteSpan>::value)
+        {
+            // Control Octet (1) + Length Octet (1) + size of a single ByteSpan
+            estimatedSize += 2 + aListAttribute[i].size();
+        }
+    }
+
+    // Encode AttributeData into a TLV Array
+    chip::Platform::ScopedMemoryBufferWithSize<uint8_t> buffer;
+    buffer.Alloc(TLV::EstimateStructOverhead(estimatedSize));
+
+    TLV::TLVWriter writer;
+    writer.Init(buffer.Get(), buffer.AllocatedSize());
+    TLV::TLVType outerContainer;
+
+    EXPECT_EQ(CHIP_NO_ERROR, writer.StartContainer(TLV::AnonymousTag(), TLV::kTLVType_Array, outerContainer));
+    for (uint8_t j = 0; j < aListAttribute.size(); j++)
+    {
+        EXPECT_EQ(CHIP_NO_ERROR, writer.Put(TLV::AnonymousTag(), aListAttribute[j]));
```

Comment:
```suggestion
    for (auto & item : aListAttribute.size())
    {
        EXPECT_EQ(CHIP_NO_ERROR, writer.Put(TLV::AnonymousTag(), item));
```

which will also fix the clang-tidy failure.

Reply:
Sure, done ( you forgot to remove .size() from `aListAttribute ` btw)